src/shared_modules/utils/rocksDBWrapper.hpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Wazuh shared modules utils
+ * Copyright (C) 2015, Wazuh Inc.
+ * September 9, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _ROCKS_DB_WRAPPER_HPP
+#define _ROCKS_DB_WRAPPER_HPP
+
+#include <iostream>
+#include <rocksdb/db.h>
+#include <string>
+
+namespace Utils
+{
+    /**
+     * @brief Wrapper class for RocksDB.
+     *
+     */
+    class RocksDBWrapper
+    {
+    public:
+        explicit RocksDBWrapper(const std::string& dbPath)
+        {
+            rocksdb::Options options;
+            options.create_if_missing = true;
+            rocksdb::DB* dbRawPtr;
+            const auto status {rocksdb::DB::Open(options, dbPath, &dbRawPtr)};
+            if (!status.ok())
+            {
+                throw std::runtime_error("Failed to open RocksDB database");
+            }
+            // Assigns the raw pointer to the unique_ptr. When db goes out of scope, it will automatically delete the
+            // allocated RocksDB instance.
+            m_db.reset(dbRawPtr);
+        }
+
+        /**
+         * @brief Put a key-value pair in the database.
+         *
+         * @param key Key to put.
+         * @param value Value to put.
+         *
+         * @note If the key already exists, the value will be overwritten.
+         */
+        void put(const std::string& key, const std::string& value)
+        {
+            if (key.empty() || value.empty())
+            {
+                throw std::invalid_argument("Key or value is empty");
+            }
+
+            const auto status {m_db->Put(rocksdb::WriteOptions(), key, value)};
+            if (!status.ok())
+            {
+                throw std::runtime_error("Error putting data: " + status.ToString());
+            }
+        }
+
+        /**
+         * @brief Get a value from the database.
+         *
+         * @param key Key to get.
+         * @param value Value to get
+         */
+        void get(const std::string& key, std::string& value)
```

Comment:
Check if the key is empty and throw an exception.

Reply:
done